src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -178,4 +179,73 @@ private function getOrCreateDonor(
 
         return $donor;
     }
+
+    /**
+     * @unreleased
+     */
+    public function addOptRefundCheckbox(int $donationId, PaymentGatewayInterface $registeredGateway)
+    {
+        $donation = Donation::find($donationId);
+        if ($donation->gatewayId === $registeredGateway::id()) {
+            ?>
+            <div id="give-gateway-opt-refund-wrap"
+                 class="give-gateway-opt-refund give-admin-box-inside give-hidden">
+                <p>
+                    <input type="checkbox" id="give-gateway-opt-refund" name="give_gateway_opt_refund" value="1" />
+                    <label for="give-gateway-opt-refund">
+                        <?php
+                        esc_html_e(sprintf('Refund the donation at %s?', $registeredGateway->getName()), 'give');
+                        ?>
+                    </label>
+                </p>
+            </div>
+            <script>
+                if (!!document.getElementById('give-payment-status') &&
+                    1 === document.querySelectorAll('div.give-admin-box > div.give-hidden input[type="checkbox"]').length
+                ) {
+                    document.getElementById('give-payment-status').addEventListener('change', function (event) {
+                        const refundCheckbox = document.getElementById('give-gateway-opt-refund');
+
+                        if (null === refundCheckbox) {
+                            return;
```

Comment:
@glaubersilva wouldn't we need to hide this checkbox if one already exists?  We might want to be more specific about targeting the checkbox also like checking if the id attribute has the string `opt-refund` in it to make sure it's the right checkbox.

Reply:
Resolved in 1c9da987e4d771d2006e0489352deffc221ed8fb

About your question:

> wouldn't we need to hide this checkbox if one already exists?

That's not necessary because each checkbox adds its own script to handle the visibility and the framework default checkbox already starts hidden because it uses the give-hidden class.